public/locales/en/translation.json

```diff
@@ -0,0 +1,61 @@
+{
```

Comment:
I'm not sure about it. Here are some potential concerns that developers might encounter while using i18n internationalization framework:

**Complexity:** Implementing internationalization in an application, especially a large one, can add complexity to the codebase. It involves managing translations, handling locale-specific formatting, date, time, and currency issues, which can be intricate and require careful attention.

**Performance overhead:** Depending on the i18n framework and the implementation, there might be some performance overhead due to additional function calls for language switching or data formatting. For applications with strict performance requirements, this might be a concern.

**Tooling and integration:** Some i18n frameworks may require specific tooling or add-ons to manage translations efficiently. Integrating these tools and managing the translation workflow might become an overhead in the development process.

**Maintenance and updates:** If not managed properly, maintaining translations and keeping them up to date with application changes can be challenging, especially for projects with frequent updates or large numbers of languages.

**Compatibility and bugs:** Like any software, i18n frameworks can have bugs or compatibility issues with specific environments, browsers, or platforms. Developers might need to invest time in identifying and resolving these issues.

**Learning curve:** Developers who are new to internationalization concepts may need to invest time in understanding the best practices, patterns, and potential pitfalls of using i18n frameworks.

**Contextual translations:** Some languages may require different translations depending on the context in which a word or phrase is used. i18n frameworks might not handle this automatically, and developers may need to implement context-aware translations themselves.

**Limited language support:** Some i18n frameworks may not support all languages or might have limitations in handling certain scripts or writing systems, potentially excluding some user groups.

@bednarskimichal What do you think about it?

Reply:
@ikhvost @bednarskimichal 
I'm not sure about it. Here are some potential concerns that developers might encounter while using i18n internationalization framework:

Complexity: Implementing internationalization in an application, especially a large one, can add complexity to the codebase. It involves managing translations, handling locale-specific formatting, date, time, and currency issues, which can be intricate and require careful attention.

Performance overhead: Depending on the i18n framework and the implementation, there might be some performance overhead due to additional function calls for language switching or data formatting. For applications with strict performance requirements, this might be a concern.

Tooling and integration: Some i18n frameworks may require specific tooling or add-ons to manage translations efficiently. Integrating these tools and managing the translation workflow might become an overhead in the development process.

Maintenance and updates: If not managed properly, maintaining translations and keeping them up to date with application changes can be challenging, especially for projects with frequent updates or large numbers of languages.

Compatibility and bugs: Like any software, i18n frameworks can have bugs or compatibility issues with specific environments, browsers, or platforms. Developers might need to invest time in identifying and resolving these issues.

Learning curve: Developers who are new to internationalization concepts may need to invest time in understanding the best practices, patterns, and potential pitfalls of using i18n frameworks.

Contextual translations: Some languages may require different translations depending on the context in which a word or phrase is used. i18n frameworks might not handle this automatically, and developers may need to implement context-aware translations themselves.

Limited language support: Some i18n frameworks may not support all languages or might have limitations in handling certain scripts or writing systems, potentially excluding some user groups.